src/witan/sen2/return/person_level/blade/plans_placements.clj

```diff
@@ -568,46 +644,29 @@
-  ([plans-placements-on-census-dates-issues-flagged]
-   (summarise-issues plans-placements-on-census-dates-issues-flagged checks))
-  ([plans-placements-on-census-dates-issues-flagged checks']
-   (-> plans-placements-on-census-dates-issues-flagged
-       (tc/group-by [:census-date])
-       (tc/aggregate (merge (update-vals checks' :summary-fn)
-                            {:num-cyp   (comp count distinct :person-table-id)
-                             :row-count tc/row-count}))
-       (tc/order-by [:census-date])
-       (tc/update-columns [:census-date] (partial map #(.format % (java.time.format.DateTimeFormatter/ISO_LOCAL_DATE))))
-       (tc/pivot->longer (complement #{:census-date}))
-       (tc/pivot->wider :census-date :$value)
-       (tc/rename-columns {:$column :issue-key})
-       (tc/map-columns :idx [:issue-key] (merge (update-vals checks' :idx)
-                                                {:num-cyp   200
-                                                 :row-count 300}))
-       (tc/map-columns :label [:issue-key] (merge (update-vals checks' :label)
-                                                  {:num-cyp   "TOTAL number of CYP"
-                                                   :row-count "TOTAL number of records"}))
-       (tc/map-columns :summary-label [:issue-key] (merge (update-vals checks' :summary-label)
-                                                          {:num-cyp   "#CYP"
-                                                           :row-count "#rows"}))
-       (tc/map-columns :desc [:issue-key] (merge (update-vals checks'
-                                                              #(apply format "%03d: %s (%s)"
-                                                                      ((juxt :idx :label :summary-label) %)))
-                                                 {:num-cyp   "(#CYP)"
-                                                  :row-count "(#rows)"}))
-       (tc/order-by [(comp (merge (update-vals checks' :idx)
-                                  {:num-cyp   200
-                                   :row-count 300})
-                           :issue-key)])
-       (tc/reorder-columns [:issue-key :idx :label :summary-label])
-       (tc/drop-columns [:issue-key :desc])
-       (tc/rename-columns {:issue-key     "key"
-                           :idx           "Index"
-                           :label         "Issue | TOTAL"
-                           :summary-label "Summary"
-                           :desc          "Index: Issue (summary statistic)"}))))
-
+  "Summarise issues flagged in `ds` as a result of running `checks`."
+  [ds checks]
+  (-> ds
+      (tc/group-by [:census-date])
+      ((fn [ds] (tc/aggregate ds (update-vals (select-keys checks (concat (tc/column-names ds)
+                                                                          (keys checks-totals)))
+                                              :summary-fn))))
```

Comment:
I might have grabbed the ds at the top in a let rather than doing the fn here, but it is fine

Reply:
Do you mean:
```
  (let [ds-column-names (tc/column-names ds)]
    (-> ds
        (tc/group-by [:census-date])
        (tc/aggregate (update-vals (select-keys checks (concat ds-column-names
                                                               (keys checks-totals)))
                                   :summary-fn))
        (tc/order-by [:census-date])
        …
```

Reply:
I like your way of doing it here and it works. It just wouldn't be what I'd reach for first.

Reply:
OK. Will leave as is. Preference noted. Thanks.